awx/ui/client/features/applications/list-applications.controller.js

```diff
@@ -0,0 +1,146 @@
+/** ***********************************************
+ * Copyright (c) 2018 Ansible, Inc.
+ *
+ * All Rights Reserved
+ ************************************************ */
+function ListApplicationsController (
+    $filter,
+    $scope,
+    $state,
+    Alert,
+    Dataset,
+    ProcessErrors,
+    Prompt,
+    resolvedModels,
+    strings,
+    Wait,
+) {
+    const vm = this || {};
+    const application = resolvedModels;
+
+    vm.strings = strings;
+    vm.activeId = $state.params.application_id;
+
+    $scope.canAdd = application.options('actions.POST');
+
+    // smart-search
+    const name = 'applications';
+    const iterator = 'application';
+    const key = 'application_dataset';
+
+    $scope.list = { iterator, name, basePath: 'applications' };
+    $scope.collection = { iterator };
+    $scope[key] = Dataset.data;
+    vm.applicationsCount = Dataset.data.count;
+    $scope[name] = Dataset.data.results;
+    $scope.$on('updateDataset', (e, dataset) => {
+        $scope[key] = dataset;
+        $scope[name] = dataset.results;
+        vm.applicationsCount = dataset.count;
+    });
+
+    vm.getModified = app => {
+        const modified = _.get(app, 'modified');
+
+        if (!modified) {
+            return undefined;
+        }
+
+        let html = $filter('longDate')(modified);
+
+        const { username, id } = _.get(app, 'summary_fields.modified_by', {});
+
+        if (username && id) {
+            html += ` by <a href="/#/users/${id}">${$filter('sanitize')(username)}</a>`;
+        }
+
+        return html;
+    };
+
+    vm.deleteApplication = app => {
+        if (!app) {
+            Alert(strings.get('error.DELETE'), strings.get('alert.MISSING_PARAMETER'));
+            return;
+        }
+
+        application.getDependentResourceCounts(application.id)
+            .then(counts => displayApplicationDeletePrompt(app, counts));
+    };
+
+    function createErrorHandler (path, action) {
+        return ({ data, status }) => {
+            const hdr = strings.get('error.HEADER');
+            const msg = strings.get('error.CALL', { path, action, status });
+            ProcessErrors($scope, data, status, null, { hdr, msg });
+        };
+    }
+
+    function handleSuccessfulDelete (app) {
+        const { page } = _.get($state.params, 'application_search');
+        let reloadListStateParams = null;
+
+        if ($scope.applications.length === 1 && !_.isEmpty(page) && page !== '1') {
+            reloadListStateParams = _.cloneDeep($state.params);
+            const pageNumber = (parseInt(reloadListStateParams.application_search.page, 0) - 1);
+            reloadListStateParams.application_search.page = pageNumber.toString();
+        }
+
+        if (parseInt($state.params.application_id, 0) === app.id) {
+            $state.go('applications', reloadListStateParams, { reload: true });
+        } else if (parseInt($state.params.application_id, 0) === application.id) {
```

Comment:
You might not need this second clause - It makes sense for `app` but `application` is a model instance so I don't think it'll ever have an `id` attribute; If I'm understanding this all correctly you shouldn't need it.